InMemoryFileSys/Directory.cs

```diff
@@ -10,139 +12,124 @@ public class Directory : IDirectory
-    
+    // Singleton instance
+    private static readonly Lazy<Directory> _root = 
+        new(() => new Directory(rootName, new SystemClock(), null));
+
     private readonly IClock _clock;
 
-    private Directory(string? name, IClock clock)
+    private readonly Directory? _parentDirectory;
+    
+    private const string slash = "/";
+
+    private const string rootName = "/";
+    
+    private Directory(string? name, IClock clock, Directory? parentDirectory)
     {
         _clock = clock ?? throw new ArgumentNullException(nameof(clock));
 
         Name = name;
         CreationDate = _clock.UtcNow;
-        _entries = new List<IFileSystemEntry>();
+        
+        _parentDirectory = parentDirectory;
+
+        if (parentDirectory == null)
+        {
+            Path = name;
+        }
+        else
+        {
+            Path = parentDirectory.Path.EndsWith(slash) ? parentDirectory.Path + name : parentDirectory.Path + slash + name;
+        }
+
+        _entries = new Dictionary<string, IFileSystemEntry>();
     }
 
     /// <inheritdoc/>
-    public void AddEntry(string name, bool isFile)
+    public IFile CreateFile(string fileName)
     {
-        if (ContainsEntry(name))
-            return;
+        // Check if the file name is null, empty, or contains slashes (which are not allowed)
+        if (string.IsNullOrWhiteSpace(fileName) || fileName.Contains(slash))
+            throw new ArgumentException("File name cannot be null, empty, or contain directory separators.");
 
-        string?[] parts = name.Split('/');
-        if (parts.Length == 1 && isFile)
+        // Check if the file already exists in this directory
+        if (_entries.ContainsKey(fileName))
         {
-            _entries.Add(CreateFile(name));
-            return;
+            throw new ArgumentException("A file with the same name already exists in this directory.");
         }
 
-        var currentDir = this;
-        foreach (var part in parts)
+        // Create the new file
+        var file = new File(Path + slash + fileName)
         {
-            if (string.IsNullOrEmpty(part))
-                continue;
+            Name = fileName,
+            CreationDate = _clock.UtcNow,
+            // Assuming a default size or some means to set the size here
+        };
 
-            var subDir = currentDir.GetSubDirectory(part);
-            if (subDir == null)
-            {
-                if (!isFile)
-                {
-                    currentDir._entries.Add(CreateDirectory(part));
-                }
-                else
-                {
-                    currentDir._entries.Add(CreateFile(part));
-                }
-            }
-            currentDir = subDir;
-        }
+        // Add the file to the current directory's entries
+        _entries[fileName] = file;
+        UpdateSize(file.Size ?? 0);  // Update the size of the directory, if applicable
```

Comment:
When you create a new file it is empty according to your logic. So, in that case we do not need to update size